pkg/build/generator/generator.go

```diff
@@ -239,6 +301,35 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from *kap
 	}
 }
 
+// resolveImageSecret looks up the Secrets provided by the Service Account and
+// attempt to find a best match for given image.
+func (g *BuildGenerator) resolveImageSecret(ctx kapi.Context, secrets []kapi.Secret, imageRef *kapi.ObjectReference, buildNamespace string) *kapi.LocalObjectReference {
+	if len(secrets) == 0 || imageRef == nil {
+		return nil
+	}
+	emptyKeyring := credentialprovider.BasicDockerKeyring{}
+	for _, secret := range secrets {
+		keyring, err := credentialprovider.MakeDockerKeyring([]kapi.Secret{secret}, &emptyKeyring)
```

Comment:
@liggitt not necessary, MakeDockerKeyring is taking care of that: https://github.com/GoogleCloudPlatform/kubernetes/blob/181124bc9b87896652b7b1bdab2e956fd024ab64/pkg/credentialprovider/keyring.go#L196